sender.go

```diff
@@ -61,25 +64,47 @@ const (
 	tlsInsecureSkipVerify tlsMode = 2
 )
 
+type transportProtocol string
+
+const (
+	protocolHttp transportProtocol = "http"
+	protocolTcp  transportProtocol = "tcp"
+)
+
 // LineSender allows you to insert rows into QuestDB by sending ILP
-// messages.
+// messages over HTTP(S)/TPC(S).
 //
 // Each sender corresponds to a single TCP connection. A sender
 // should not be called concurrently by multiple goroutines.
 type LineSender struct {
```

Comment:
Keeping this struct as 1-in-2 makes it harder to maintain. TCP and HTTP senders have different behavior, e.g. around retry, so it's better to split them. I suggest introducing `HttpLineSender` struct for ILP/HTTP and exposing the following constructor functions: `NewLineSender`, `NewLineSenderFromConf`, `NewHttpLineSender`, `NewHttpLineSenderFromConf`. This way we won't break existing client code and keep implementations separate. Also, buffer related stuff could be extracted into a common struct which would be embedded into both `LineSender` and `HttpLineSender` - this way we'll avoid code duplication.

Reply:
> NewHttpLineSenderFromConf

I don't think we need this since we specify the protocol in the conf string itself. Maybe `NewLineSenderFromConf` is enough? And possible alias `NewLineSender` to `NewTcpLineSender` for clarity? (while keeping the existing function so as not to break compatibility)

Reply:
It won't be possible to return one of two different structs from a single function without an interface. That's why we'd need different FromConf functions.